internal/component/database_observability/mysql/collector/health_check.go

```diff
@@ -0,0 +1,356 @@
+	r := healthCheckResult{name: "DBConnectionValid"}
+	subCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	if err := db.PingContext(subCtx); err != nil {
+		r.value = err.Error()
+		return r
+	}
+	r.result = true
+	r.value = "ok"
+	return r
+}
+
+// checkMySQLVersion validates that the database the MySQL version >= 8.0.
+func checkMySQLVersion(ctx context.Context, db *sql.DB) healthCheckResult {
+	r := healthCheckResult{name: "MySQLVersionSupported"}
+	const q = `SELECT VERSION()`
+	var version string
+	if err := db.QueryRowContext(ctx, q).Scan(&version); err != nil {
+		r.err = fmt.Errorf("query version(): %w", err)
+		return r
+	}
+
+	matches := mysqlVersionRegex.FindStringSubmatch(version)
+	if len(matches) <= 1 {
+		r.err = fmt.Errorf("unexpected version format: %s", version)
+		return r
+	}
+
+	parsed, err := semver.ParseTolerant(matches[1])
+	if err != nil {
+		r.err = fmt.Errorf("parse semver: %w", err)
+		return r
+	}
+
+	r.result = semver.MustParseRange(">=8.0.0")(parsed)
+	r.value = version
+	return r
+}
+
+// checkAlloyVersion reports the running Alloy version.
+func checkAlloyVersion(ctx context.Context, db *sql.DB) healthCheckResult {
+	r := healthCheckResult{name: "AlloyVersion"}
+	// Always succeeds; returns the version string embedded at build time.
+	r.result = true
+	r.value = build.Version
+	return r
+}
+
+// checkRequiredGrants verifies required privileges are present.
+func checkRequiredGrants(ctx context.Context, db *sql.DB) healthCheckResult {
+	r := healthCheckResult{name: "RequiredGrantsPresent"}
+	req := map[string]bool{
+		"PROCESS":            false,
+		"REPLICATION CLIENT": false,
+		"SELECT":             false,
+		"SHOW VIEW":          false,
+	}
+
+	rows, err := db.QueryContext(ctx, "SHOW GRANTS")
+	if err != nil {
+		r.err = fmt.Errorf("SHOW GRANTS: %w", err)
+		return r
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var grantLine string
+		if err := rows.Scan(&grantLine); err != nil {
+			r.err = fmt.Errorf("scan SHOW GRANTS: %w", err)
+			return r
+		}
+		up := strings.ToUpper(grantLine)
+
+		// Mark individual privileges if present on *.* scope.
+		for k := range req {
+			if strings.Contains(up, " ON *.*") && strings.Contains(up, k) {
+				req[k] = true
+			}
+		}
+	}
+	if err := rows.Err(); err != nil {
+		r.err = fmt.Errorf("iterate SHOW GRANTS: %w", err)
+		return r
+	}
+
+	r.result = true
+	for k, found := range req {
+		if !found {
+			r.result = false
+			if r.value == "" {
+				r.value = "missing: " + k
+			} else {
+				r.value += "," + k
+			}
+		}
+	}
+
+	return r
+}
+
+// checkDigestVariablesLength ensures text/digest length variables are >= 4096.
+func checkDigestVariablesLength(ctx context.Context, db *sql.DB) healthCheckResult {
+	r := healthCheckResult{name: "DigestVariablesLengthCheck"}
+	const q = `
+SELECT
+	@@performance_schema_max_sql_text_length,
+	@@performance_schema_max_digest_length,
+	@@max_digest_length`
```

Comment:
Could we use perhaps these?

mysql_global_variables_performance_schema_max_digest_length
mysql_global_variables_performance_schema_max_sql_text_length
mysql_global_variables_max_digest_length